client/v2/tx/tx.go

```diff
@@ -150,66 +166,85 @@ func SimulateTx(ctx client.Context, flagSet *pflag.FlagSet, msgs ...transaction.
 // BroadcastTx attempts to generate, sign and broadcast a transaction with the
 // given set of messages. It will also simulate gas requirements if necessary.
 // It will return an error upon failure.
-func BroadcastTx(clientCtx client.Context, txf Factory, broadcaster broadcast.Broadcaster, msgs ...transaction.Msg) error {
+func BroadcastTx(
+	ctx context.Context,
+	txf Factory,
+	broadcaster broadcast.Broadcaster,
+	skipConfirm bool,
+	msgs ...transaction.Msg,
+) ([]byte, error) {
 	if txf.simulateAndExecute() {
 		err := txf.calculateGas(msgs...)
 		if err != nil {
-			return err
+			return nil, err
 		}
 	}
 
 	err := txf.BuildUnsignedTx(msgs...)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if !clientCtx.SkipConfirm {
+	if !skipConfirm {
 		encoder := txf.txConfig.TxJSONEncoder()
 		if encoder == nil {
-			return errors.New("failed to encode transaction: tx json encoder is nil")
+			return nil, errors.New("failed to encode transaction: tx json encoder is nil")
 		}
 
 		unsigTx, err := txf.getTx()
 		if err != nil {
-			return err
+			return nil, err
 		}
 		txBytes, err := encoder(unsigTx)
 		if err != nil {
-			return fmt.Errorf("failed to encode transaction: %w", err)
+			return nil, fmt.Errorf("failed to encode transaction: %w", err)
 		}
 
-		if err := clientCtx.PrintRaw(txBytes); err != nil {
-			_, _ = fmt.Fprintf(os.Stderr, "error: %v\n%s\n", err, txBytes)
+		clientCtx, err := clientcontext.ClientContextFromGoContext(ctx)
```

Comment:
Yeah so broadcast tx shouldn't print anything as it returns the output. We should let the caller (which is autocli or any other caller) to handle the printing.

Reply:
I know this command is somewhat interactive, as it takes an user input, but that should be handled by something else (autocli or client) imho not here.

Reply:
We could have an helper for generate, and one for sign and broadcast and kill all that input logic. In autocli we can easily handle the printing + sign+broadcast in two seperate call.